tests/python/tests/base.py

```diff
@@ -93,9 +94,19 @@ class TestCaseWithFlask(TestCase):
     def setUpClass(cls):
         cls.server = Process(target=application, args=(cls._TS_PORT,))
         cls.server.start()
-        time.sleep(0.5)
         cls.MOCKURL = MOCKURL_TEMPLATE % cls._TS_PORT.value
         cls.PORT = cls._TS_PORT.value
+        # Wait for the server to start (max 5 seconds)
+        for i in range(50):
+            time.sleep(0.1)
+            try:
+                requests.get(cls.MOCKURL)
```

Comment:
There should be a timeout set for requests.get() to prevent hanging:
http://docs.python-requests.org/en/master/user/quickstart/#timeouts

Reply:
Thanks, I added that.